### PR TITLE
Create IpPacketRouter

### PR DESCRIPTION
**Pull request by octol**
# Description

Create IpPacketRouter

### CODE REVIEW

File: common/wireguard/src/platform/linux/tun_device.rs [hunk 1]
```diff
@@ -123,7 +123,7 @@ impl TunDevice {
 
         // This is how wireguard does it, by consulting the AllowedIPs table.
         if false {
-            let peers = self.peers_by_ip.lock().await;
+            let peers = self.peers_by_ip.as_ref().unwrap().lock().await;
```
Comment (jstuczyn) [inline common/wireguard/src/platform/linux/tun_device.rs]: if you're just always unwrapping it, what's the point of wrapping it in an option in the first place?

Comment (octol) [reply common/wireguard/src/platform/linux/tun_device.rs]: Well this is draft so ... ;)
Hold your horses

Comment (jstuczyn) [reply common/wireguard/src/platform/linux/tun_device.rs]: 🛑 🏇🏻

Comment (octol) [reply common/wireguard/src/platform/linux/tun_device.rs]: There we go, no more unwrap :)

**Pull request merged**